### PR TITLE
fix: added test case and working solution for empty li elements in the 'Create an ordered list' challenge.

### LINKED ISSUE #18007: Ordered and Unordered list can be empty
Probably not a big deal, but in the ordered and unordered list session the tests pass even if the list items are empty

https://learn.freecodecamp.org/responsive-web-design/basic-html-and-html5/create-a-bulleted-unordered-list

* Browser Name: Chrome
* Browser Version: 69.0.3497.32 (Official Build) beta (64-bit)
* Operating System: Mac OS High Seirra

screen shot 2018-08-12 at 11 25 26

### PR DESCRIPTION
**Pull request by ghost**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #18007

### CODE REVIEW

**Comment by alan-nguy** [comment]
This is my very first PR.  Apologies in advance if PR wasn't done correctly. This is for issue https://github.com/freeCodeCamp/freeCodeCamp/issues/18007

**Review by theNaut1lus — CHANGES_REQUESTED** [review]
@alan-nguy 
Please add an error message which will be displayed in the console, whenever a user might fail this test.
follow along with how other tests are made.

**Comment by co16353sidak** [comment]
@alan-nguy 

Good Work for a first PR, I would suggest you create an issue first for this as this seems like something which would be better discussed first.

Straight Pull Requests are only desirable when we are fixing issues not really needing any discussion however we are changing a test in a challenge here so it could use some discussion first.

you can then link the PR with this issue.

Awesome work for your first PR 👍

**Comment by alan-nguy** [comment]
@co16353sidak 
Thank you for for the pointer. A new PR has been submitted.
This PR is for issue https://github.com/freeCodeCamp/freeCodeCamp/issues/18007

**Comment by RandellDawson** [comment]
@alan-nguy I tested your solution it seems to work well.  One thing this challenge is missing (not your fault) is an actual valid working solution.  When one is present it is used behind-the-scenes to validate all the written test case still pass with a working solution. Replace the section which looks like:
```
## Solution
<section id='solution'>

````js
// solution required
````
</section>
```
with
```
## Solution
<section id='solution'>

````html
<h2>CatPhotoApp</h2>
<main>
  <p>Click here to view more <a href="#">cat photos</a>.</p>

  <a href="#"><img src="https://bit.ly/fcc-relaxing-cat" alt="A cute orange cat lying on its back."></a>

  <ul>
    <li>milk</li>
    <li>mice</li>
    <li>catnip</li>
  </ul>
</main>```
</section>
```

**Comment by RandellDawson** [comment]
Hi @alan-nguy,

So we have a unique situation here that I am hoping you will not be to upset about.  Even though you created your PR before any other PR was created for issue #18007, another user was given the go ahead to create a PR.  Most of the time, when users say they will create a PR, they never do.  However, in this case, the user did create a valid additional test.  It is not the same as yours, but it works all the same. 

Seeing that the other user was given "consent" to work on this, I have approved the other PR instead for the additional test.  However, the other PR did not provide a working solution (as I requested you add), so if you can just remove the additional test off this PR and update the PR title with applicable wording, this PR can still be merged independently of the other one.

Also, neither yours nor the other PR added the additional test to the [Basic HTML and HTML5: Create an Ordered List](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-an-ordered-list.english.md) challenge, so the other user agreed to let you be the one to add a similar test for that challenge and include it on this PR.  Go ahead an add a working solution for that challenge to this PR and that will truly close out issue #18007.

Sorry for any confusion or frustration this may have caused you, but this is kind of a unique situation that does not come up very often.  If you have any questions about anything I have stated here, please let me know.

**Comment by RandellDawson** [comment]
**ATTENTION REVIEWERS:**  Please do not merge this PR before #35234 has been merged.

**Comment by alan-nguy** [comment]
@RandellDawson Thank you for the update. Nothing to be upset about. PR coming in shortly in the next hour. Cheers!

**Review by RandellDawson — APPROVED** [review]
On my local machine, I have tested both solutions for both challenges and confirmed the new tests for the `Create an ordered list` work as they should.

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-an-ordered-list.english.md
Comment (ezioda004) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-an-ordered-list.english.md]: Don't need double negation/NOT (`!!`). `.length` will be a valid number and only falsy value could be `0` which is falsy anyway.

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-an-ordered-list.english.md
Comment (ezioda004) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-an-ordered-list.english.md]: Likewise here and it should be `ordered list` instead of `unordered list`.

**Review by ezioda004 — CHANGES_REQUESTED** [review]
Hi @alan-nguy,

Please check my review.
Thanks for contributing!

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-an-ordered-list.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-an-ordered-list.english.md]: @ezioda004 Look at the other tests above.  They have tests for both ordered and unordered lists.

Comment (ezioda004) [reply curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-an-ordered-list.english.md]: @RandellDawson Yes ofcourse but in this case the **text** and the *testString* don't match.
>**text**: The <code>li</code> elements in your *ordered* list should not be empty.
>**testString**: Your <code>li</code> elements in your *unordered* list should not be empty.

Comment (RandellDawson) [reply curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-an-ordered-list.english.md]: @ezioda004 - I assume your are talking about the last text and testString.   I was focused on the code and not the `text`.  Great catch!  That is why it is always good to have multiple reviews on these challenges.

**Comment by alan-nguy** [comment]
@ezioda004 thnx for the review! New PR added.

**Comment by RandellDawson** [comment]
@ezioda004 I pull this PR down again with the most recent changes and test it locally.  I think this PR is ready to merge.

**Review by Manish-Giri — COMMENTED** [review]
Tested locally, and the changes do account for empty list items.

A few observations though -

For the _**create-a-bulleted-unordered-list**_ challenge:

- "**Remove the last two p elements** and create an unordered list" : This doesn't really affect the challenge outcome, as the tests pass even with the `p` elements present. As the challenge is about teaching lists, I propose removing the text "Remove the last two p elements" from the challenge description. This can be done in a separate PR too.

- A space is allowed in the opening `ol`/`li` tag, but not in the closing tag. So, for instance, this list item `<li >jk</li>` passes the test, but `<li >jk</li >` doesn't. If we're allowing spaces in the opening `<li>`, then the same should go for closing `</li>` too, for consistency - `assert(code.match(/<\/li>/gi) && code.match(/<li[\s>]/gi)`. Same for the `ol` tags.

For the _**create-an-ordered-list**_ challenge:

- No space is permitted in either the opening or closing `ul`/`ol`/`li` tags. This seems to contradict the earlier challenge where a space in the opening tags is permitted. To be consistent, both challenges should ideally follow the same behavior.

Again, since these items are not directly related to the issue in hand, they can be dealt with in another PR.

**Comment by RandellDawson** [comment]
@Manish-Giri I think all of your observations are valid.  I think a new issue with these items should be created and then we can discuss all options of how best to deal with them before even creating a PR.

In several of my latest PRs, I have been stripping white space from the code (where it does not affect the code's purpose).  My regular expressions become much simpler in many cases, because I don't have to add all possible `\s*` just to account for how someone white attempt to write the code.  Also, I really believe we could get away from the heavy use of regular expression in the asserts if we rethink how we can check for valid solutions.  I have a few ideas that we can discuss.  One is using a minifier for CSS and HTML sections which would be used to before the test so in the case where we are looking for very specific CSS in the code, we at least don't have to deal with trying to account for all the possible white space that could be present and still be valid code.

**Comment by RandellDawson** [comment]
@ezioda004 I guess we are just waiting for review to validate the changes you requested have been met in your opinion, before merging this PR.

**Comment by RandellDawson** [comment]
@ezioda004 Any change for a final review?

**Review by ezioda004 — APPROVED** [review]
LGTM!

**Comment by ezioda004** [comment]
Thank you @alan-nguy for your contribution. Looking forward to more contributions from you! 🎉

**Pull request merged**
